server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropSetTool.java

```diff
@@ -0,0 +1,391 @@
+      throw new IllegalStateException(ex);
+    }
+  }
+
+  private PropStoreKey<?> getPropKey(final InstanceId iid, final ZooPropSetTool.Opts opts,
+      final ZooReader zooReader) {
+
+    // either tid or table name option provided, get the table id
+    if (!opts.tableOpt.isEmpty() || !opts.tableIdOpt.isEmpty()) {
+      TableId tid = getTableId(iid, opts, zooReader);
+      return TablePropKey.of(iid, tid);
+    }
+
+    // either nid of namespace name provided, get the namespace id.
+    if (!opts.namespaceOpt.isEmpty() || !opts.namespaceIdOpt.isEmpty()) {
+      NamespaceId nid = getNamespaceId(iid, opts, zooReader);
+      return NamespacePropKey.of(iid, nid);
+    }
+
+    // no table or namespace, assume system.
+    return SystemPropKey.of(iid);
+  }
+
+  private TableId getTableId(final InstanceId iid, final ZooPropSetTool.Opts opts,
+      final ZooReader zooReader) {
+    if (!opts.tableIdOpt.isEmpty()) {
+      return TableId.of(opts.tableIdOpt);
+    }
+    Map<NamespaceId,String> nids = getNamespaceIdToNameMap(iid, zooReader);
+
+    Map<TableId,String> tids = getTableIdToName(iid, nids, zooReader);
+    return tids.entrySet().stream().filter(entry -> opts.tableOpt.equals(entry.getValue()))
+        .map(Map.Entry::getKey).findAny()
+        .orElseThrow(() -> new IllegalArgumentException("Could not find table " + opts.tableOpt));
+  }
+
+  private NamespaceId getNamespaceId(final InstanceId iid, final ZooPropSetTool.Opts opts,
+      final ZooReader zooReader) {
+    if (!opts.namespaceIdOpt.isEmpty()) {
+      return NamespaceId.of(opts.namespaceIdOpt);
+    }
+    Map<NamespaceId,String> nids = getNamespaceIdToNameMap(iid, zooReader);
+    return nids.entrySet().stream().filter(entry -> opts.namespaceOpt.equals(entry.getValue()))
+        .map(Map.Entry::getKey).findAny().orElseThrow(
+            () -> new IllegalArgumentException("Could not find namespace " + opts.namespaceOpt));
+  }
+
+  static class Opts extends ConfigOpts {
+
+    @Parameter(names = {"-d", "--delete"}, description = "delete a property")
+    public String deleteOpt = "";
+    @Parameter(names = {"-f", "--filter"},
+        description = "show only properties that contain this string in their name.")
+    public String filterOpt = "";
+    @Parameter(names = {"-fv", "--filter-with-values"},
+        description = "show only properties that contain this string in their name.")
+    public String filterWithValuesOpt = "";
+    @Parameter(names = {"--instanceName"},
+        description = "Specify the instance name to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceName = "";
+    @Parameter(names = {"--instanceId"},
+        description = "Specify the instance id to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceId = "";
```

Comment:
We don't need these. We previously dropped a lot of these complicated CLI options for these kind of server-side utils in favor of just setting the path to the server-side config file, which already has everything that is needed. I didn't realize you had added similar options to the ZooInfoViewer tool, and would recommend they be dropped from there as well, to keep things simple. Anybody using a server-side util should already have a server-side config file with all the needed connectionString information.

Reply:
Removed in d11c27ea95